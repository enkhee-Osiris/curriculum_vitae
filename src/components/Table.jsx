function Table({ data }) {
  return (
    <table className="w-full overflow-x-auto">
      <tbody className="divide-y divide-black dark:divide-gray-500">
        {Object.keys(data).map((key) => (
          <tr key={`table_${key.toLowerCase()}`}>
            <td className="py-2 font-medium">{key}</td>
            <td className="py-2">{data[key]}</td>
          </tr>
        ))}
      </tbody>
    </table>
  );
}

export default Table;
