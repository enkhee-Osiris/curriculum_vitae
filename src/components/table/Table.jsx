import React from 'react';
import { objectOf, string } from 'prop-types';

const Table = ({ datas }) => {
  const rows = Object.keys(datas).map(key => (
    <tr key={key}>
      <td style={{ fontWeight: 500 }}>{key}</td>
      <td>{datas[key]}</td>
    </tr>));

  return (
    <table className="u-full-width">
      <tbody>
        {rows}
      </tbody>
    </table>
  );
};

Table.propTypes = {
  datas: objectOf(string).isRequired,
};

export { Table as default };
