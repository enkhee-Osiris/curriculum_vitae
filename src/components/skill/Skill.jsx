import React, { Fragment } from 'react';
import { objectOf, string } from 'prop-types';
import Table from '../table/Table';

const Skill = ({ datas }) => (
  <Fragment>
    <h3>Skills</h3>
    <Table datas={datas} />
  </Fragment>
);

Skill.propTypes = {
  datas: objectOf(string).isRequired,
};

export { Skill as default };
