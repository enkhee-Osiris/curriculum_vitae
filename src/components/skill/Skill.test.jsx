import React from 'react';
import { shallow } from 'enzyme';
import Skill from './Skill';
import Table from '../table/Table';

describe('<Skill />', () => {
  const props = {
    datas: {
      Type: 'test',
    },
  };

  it('should be defined', () => {
    expect(Skill).toBeDefined();
  });

  it('renders correctly', () => {
    const skill = shallow(<Skill {...props} />);

    expect(skill).toMatchSnapshot();
  });

  it('renders h3', () => {
    const skill = shallow(<Skill {...props} />);
    const title = <h3>Skills</h3>;

    expect(skill.contains(title)).toEqual(true);
  });

  it('renders Table', () => {
    const skill = shallow(<Skill {...props} />);
    const table = <Table datas={props.datas} />;

    expect(skill.contains(table)).toEqual(true);
  });
});
