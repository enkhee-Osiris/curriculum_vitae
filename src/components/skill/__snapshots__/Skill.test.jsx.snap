// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Skill /> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Skill
    datas={
        Object {
            "Type": "test",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <h3>
          Skills
</h3>,
        <Table
          datas={
                    Object {
                              "Type": "test",
                            }
          }
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Skills",
        },
        "ref": null,
        "rendered": "Skills",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "datas": Object {
            "Type": "test",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <h3>
            Skills
</h3>,
          <Table
            datas={
                        Object {
                                    "Type": "test",
                                  }
            }
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Skills",
          },
          "ref": null,
          "rendered": "Skills",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "datas": Object {
              "Type": "test",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
